README
NAME
    Bundle::Perl::Critic - A CPAN bundle for Perl::Critic and related
    modules

SYNOPSIS
      perl -MCPAN -e 'install Bundle::Perl::Critic'

DESCRIPTION
    This is a CPAN bundle file for installing Perl::Critic,
    Test::Perl::Critic, <criticism> and several Policy plugin modules, along
    with all their dependencies.

    In addition to the distributions included below, there is
    Perl::Critic::Dynamic. This is not incorporated here because it actually
    compiles code and is therefore potentially dangerous.

    WARNING: This bundle will install Policies that directly conflict with
    each other. If you do not use a .perlcriticrc file, and your severity is
    set high enough, there is no way for your code to not have violations.

CONTENTS
        Perl::Critic

        Test::Perl::Critic

        Test::Perl::Critic::Progressive

        Perl::Critic::More

        Perl::Critic::Bangs

        Perl::Critic::Compatibility

        Perl::Critic::Lax

        Perl::Critic::Nits

        Perl::Critic::PetPeeves::JTRAMMELL

        Perl::Critic::Pulp

        Perl::Critic::Storable

        Perl::Critic::StricterSubs

        Perl::Critic::Swift

        Perl::Critic::Tics

        criticism

AUTHOR
    Jeffrey Ryan Thalhammer <thaljef@cpan.org>

COPYRIGHT
    Copyright (c) 2005-2008 Jeffrey Ryan Thalhammer. All rights reserved.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself. The full text of this license can
    be found in the LICENSE file included with this module.



Changes
#######################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/trunk/Bundle-Perl-Critic/Changes $
#     $Date: 2008-08-11 11:28:42 -0500 (Mon, 11 Aug 2008) $
#   $Author: clonezone $
# $Revision: 2657 $
########################################################################

[1.023] Released on 2008-08-11

     Added new Perl-Critic modules:

     * Perl::Critic::Storable

[1.022] Released on 2008-07-06

     Added new Perl-Critic modules:

     * Perl::Critic::PetPeeves::JTRAMMELL
     * Perl::Critic::Pulp

[1.021] Released on 2008-06-17

    "use 5.006001" in Makefile.PL and Build.PL.

[1.020] Released on 2008-06-07

     Added new Perl-Critic modules:

     * Perl::Critic::Compatibility

[1.01] Released on 2007-09-15

     Added new Perl-Critic modules:

     * Perl::Critic::Nits

     Mention Perl::Critic::Dynamic.

[1.00] Released on 2007-08-25

     Added new Perl-Critic modules:

     * Perl::Critic::Tics

[0.02] Released on 2007-07-09

     Added new Perl-Critic modules:

     * Perl::Critic::Swift
     * Perl::Critic::StricterSubs
     * Test::Perl::Critic::Progressive

[0.01] Released on 2006-11-25

     Initial release.  Contains the following modules:

     * Perl::Critic
     * Test::Perl::Critic
     * Perl::Critic::More
     * Perl::Critic::Bangs
     * Perl::Critic::Lax
     * criticism


Build.PL
#######################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/trunk/Bundle-Perl-Critic/Build.PL $
#     $Date: 2008-06-17 14:27:21 -0500 (Tue, 17 Jun 2008) $
#   $Author: clonezone $
# $Revision: 2450 $
########################################################################

use 5.006001;

use strict;
use warnings;

use Module::Build;

my $class = Module::Build->subclass( code => <<'END_SUBCLASS' );

    sub ACTION_test {
        my ($self) = @_;

        $self->depends_on('manifest');

        return $self->SUPER::ACTION_test();
    }

    sub ACTION_authortest {
        my ($self) = @_;

        $self->depends_on('build');
        $self->depends_on('manifest');
        $self->depends_on('distmeta');

        # No xt/author here.
        $self->test_files( qw< t > );
        $self->recursive_test_files(1);

        $self->depends_on('test');

        return;
    }

    sub ACTION_distdir {
        my ($self) = @_;

        $self->depends_on('authortest');

        return $self->SUPER::ACTION_distdir();
    }

END_SUBCLASS

my $builder = $class->new(
    module_name     => 'Bundle::Perl::Critic',
    dist_author     => 'Jeffrey Thalhammer <thaljef@cpan.org>',
    dist_abstract   => 'A CPAN bundle for Perl::Critic and related modules.',
    license         => 'perl',
    dynamic_config  => 1,
    create_readme   => 1,
    create_packlist => 1,
    sign            => 0,
    meta_merge      => {
        resources   => {
            homepage    => 'http://perlcritic.com',
            license     => 'http://dev.perl.org/licenses/',
            Repository  => 'http://perlcritic.tigris.org/svn/perlcritic/trunk/Bundle-Perl-Critic/',
            MailingList => 'http://perlcritic.tigris.org/servlets/SummarizeList?listName=users',
        }
    },
);

$builder->create_build_script();


lib/Bundle/Perl/Critic.pm
#######################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/trunk/Bundle-Perl-Critic/lib/Bundle/Perl/Critic.pm $
#     $Date: 2008-08-11 11:28:42 -0500 (Mon, 11 Aug 2008) $
#   $Author: clonezone $
# $Revision: 2657 $
#        ex: set ts=8 sts=4 sw=4 expandtab :
########################################################################

package Bundle::Perl::Critic;

use 5.006001;

use strict;
use warnings;

our $VERSION = '1.023';

1;

__END__

#-----------------------------------------------------------------------------

=pod

=for stopwords CPAN Thalhammer

=head1 NAME

Bundle::Perl::Critic - A CPAN bundle for Perl::Critic and related modules

=head1 SYNOPSIS

  perl -MCPAN -e 'install Bundle::Perl::Critic'

=head1 DESCRIPTION

This is a CPAN bundle file for installing L<Perl::Critic>,
L<Test::Perl::Critic>, <criticism> and several Policy plugin modules,
along with all their dependencies.

In addition to the distributions included below, there is
L<Perl::Critic::Dynamic>.  This is not incorporated here because it
actually compiles code and is therefore potentially dangerous.

B<WARNING>: This bundle will install Policies that directly conflict
with each other.  If you do not use a F<.perlcriticrc> file, and your
severity is set high enough, there is no way for your code to not have
violations.

=head1 CONTENTS

    Perl::Critic

    Test::Perl::Critic

    Test::Perl::Critic::Progressive

    Perl::Critic::More

    Perl::Critic::Bangs

    Perl::Critic::Compatibility

    Perl::Critic::Lax

    Perl::Critic::Nits

    Perl::Critic::PetPeeves::JTRAMMELL

    Perl::Critic::Pulp

    Perl::Critic::Storable

    Perl::Critic::StricterSubs

    Perl::Critic::Swift

    Perl::Critic::Tics

    criticism

=head1 AUTHOR

Jeffrey Ryan Thalhammer <thaljef@cpan.org>

=head1 COPYRIGHT

Copyright (c) 2005-2008 Jeffrey Ryan Thalhammer.  All rights reserved.

This program is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.  The full text of this license
can be found in the LICENSE file included with this module.

=cut
